ramp4-pcar4/storylines-editor
# Redesign product details form

### Related Item(s)
Issues:
#400
#330 

### Changes
- [FEATURE] Implement new metadata details form, with different settings/available actions for new storylines, existing storylines, and modal-based editing.
- [FIX] Various page formatting fixes.

### Notes
_**New storyline:**_
![image](https://github.com/user-attachments/assets/088e8702-5493-41c2-aef9-ec6a3ad0e1c8)

_**Existing storyline, editing not selected:**_
![image](https://github.com/user-attachments/assets/43be8597-f974-4e40-a24f-ad455dd6ad41)

_**Existing storyline, editing selected:**_
![image](https://github.com/user-attachments/assets/b4b2a9e3-8fe3-427d-ab80-420c112777f2)

**_Modal editing:_**
<img width="1470" alt="image" src="https://github.com/user-attachments/assets/31388e10-7f0e-4845-ba67-8ce5b4475bc2">

**_Mobile metadata page(s):_**
![image](https://github.com/user-attachments/assets/7afc8110-e302-4ce6-b7fe-a968c7f1166d)

![image](https://github.com/user-attachments/assets/f67bfb07-18d1-4b3d-8f1f-a13e6d1cd61b)



### Testing
Steps:
1. Open the editor, and enter the "Create new product" page. Play with the new form. Notice it doesn't have a preview button.
2. Open the "Edit existing product" page, use `00000000-0000-0000-0000-000000000000`. View the new populated metadata form, press the buttons to edit and preview, etc.
3. Press continue to go to the editor, then press the "Edit Project Metadata" button to open the new metadata modal.
